src/commands/infra/generate.ts

```diff
@@ -200,49 +200,133 @@ export const generateConfig = async (projectPath: string): Promise<void> => {
         If there is a match, then replace parent key-value
         If there is no match between the parent and leaf,
         then append leaf key-value parent key-value JSON */
-      for (const parentKey in parentDefinitionJSON.variables) {
-        if (parentKey) {
-          for (const leafKey in leafDefinitionJSON.variables) {
-            if (parentKey === leafKey) {
-              let parentVal = parentDefinitionJSON.variables[parentKey];
-              parentVal = leafDefinitionJSON.variables[leafKey];
-            } else {
-              // Append to parent variables block
-              const leafVal = leafDefinitionJSON.variables[leafKey];
-              parentDefinitionJSON.variables[leafKey] = leafVal;
-            }
-          }
-        }
-      }
       // Create a generated parent directory
       const parentDirectory = await createGenerated(cwdPath + "-generated");
       // Then, create generated child directory
       const childDirectory = await createGenerated(
         path.join(parentDirectory, projectPath)
       );
-      // Generate Terraform files in generated directory
-      const spkTfvarsObject = await generateSpkTfvars(
-        parentDefinitionJSON.variables
-      );
-      await checkSpkTfvars(childDirectory);
-      await writeToSpkTfvarsFile(spkTfvarsObject, childDirectory);
-      // const templatePath = await parseDefinitionJson(projectPath);
-      await copyTfTemplate(templatePath, childDirectory);
+      if (parentDefinitionJSON.variables) {
+        for (const parentKey in parentDefinitionJSON.variables) {
+          if (parentKey) {
+            for (const leafKey in leafDefinitionJSON.variables) {
+              if (parentKey === leafKey) {
+                let parentVal = parentDefinitionJSON.variables[parentKey];
+                parentVal = leafDefinitionJSON.variables[leafKey];
+              } else {
+                // Append to parent variables block
+                const leafVal = leafDefinitionJSON.variables[leafKey];
+                parentDefinitionJSON.variables[leafKey] = leafVal;
+              }
+            }
+          }
```

Comment:
nit: this is the same logic as the under `Create a backend.tfvars for remote backend configuration`, it can be a method and you can pass either `parentDefinitioinJSON.variables` or `leafDefinitionJSON.backend`.

Reply:
I had meant to do that, thanks for the reminder! :)